EIPS/eip-1715.md

```diff
@@ -0,0 +1,184 @@
+
+```
+[ version: P, signalingBits: S, clientIdentity: B ]
+```
+
+For this EIP, `version` should equal to `1`, and `signalingBits` and `clientIdentity` fulfill the rest of the extraData field as long as it does not exceed the maximum size allowed. This allows us to have way more concurrent signalings and better backward compatibility than Bitcoin.
+
+`signalingBits` are right-aligned, i.e. `0b1` has its bit at index 0 set, `0b10` has its bit at index 1 set, `0b100` has its bit at index 2 set.
+
+### Deployment states
+
+With each block and fork, we associate a deployment state. The possible states are:
+
+* **DEFINED** is the first state that each fork starts out as. The genesis block for any chain SHALL by definition be in this state for each deployment.
+* **STARTED** for blocks past the starttime.
+* **LOCKED_IN** after STARTED, if at least threshold out of windowsize blocks have the associated bit set in `signalingBits` in `extraData` block header, measured at next height that is evenly divisible by the windowsize.
+* **ACTIVE** for all blocks after the grace period conditions have been met.
+* **FAILED** if past the timeout time and LOCKED_IN was not reached.
+
+In accordance with BIP9, a block's state SHALL never depend on its own extraData; only on that of its ancestors.
+
+### Fork deployment parameters
+
+Each fork deployment is specified by the following per-chain parameters:
+
+* The **name** specifies a very brief description of the fork, reasonable for use as an identifier. For deployments described in a single BIP, it is recommended to use the name "bipN" where N is the appropriate BIP number.
+* The **bit** determines which bit in the extraData field of the block is to be used to signal the fork deployment.
+* The **start** specifies a block number at which the bit gains its meaning.
+* The **timeout** specifies a time at which the deployment is considered failed. If the current block number >= (start + timeout) and the fork has not yet locked in (including this block's bit state), the deployment is considered failed on all descendants of the block.
+* The **windowsize** specifies the number of past blocks (including the block under consideration) to be taken into account for locking in a fork.
+* The **threshold** specifies a number of blocks, in the range of 1..windowsize, which must signal for a fork in order to lock it in. The support is measured when the chain height is evenly divisible by the windowsize. If the windowsize is set to 2016 (as in BIP9) this coincides with the 2016-block re-targeting intervals.
+* The **minlockedblocks** specifies a minimum number of blocks which a fork must remain in locked-in state before it can become active. Both minlockedblocks and minlockedtime (see below) must be satisfied before a fork can become active. If the current block number >= (minlockedblocks + the block number that locked in the fork), then the fork becomes activated. 
```

Comment:
```suggestion
* The **minlockedblocks** specifies a minimum number of blocks which a fork must remain in locked-in state before it can become active. If the `current block number >= (minlockedblocks + the block number that locked in the fork)`, then the fork becomes activated.
```